test/controllers/valuations_controller_test.rb

```diff
@@ -3,17 +3,107 @@
   end
 
-  test "create" do
+  test "should create valuation" do
     assert_difference("Valuation.count") do
       post account_valuations_url(@account), params: { valuation: { value: 1, date: Date.current, type: "Appraisal" } }
     end
   end
+
+  test "should sync account after create" do
+    date = Date.current - 1.day
+    value = 2
+
+    @account.sync
+
+    assert_changes("@account.balance_on(date)", to: value) do
+      post account_valuations_url(@account), params: { valuation: { value:, date:, type: "Appraisal" } }
+      perform_enqueued_jobs
+    end
+  end
+
+  test "should do a partial account sync after create" do
+    date = Date.current - 1.day
+    @account.sync
+    @account.balances.where(date: date - 1.day).update!(balance: 200)
+
+    assert_no_changes("@account.balance_on(date - 1.day)") do
+      post account_valuations_url(@account), params: { valuation: { value: 1, date:, type: "Appraisal" } }
+      perform_enqueued_jobs
+    end
+  end
+
+  test "should update valuation" do
+    date = @valuation.date
+    patch valuation_url(@valuation), params: { valuation: { account_id: @valuation.account_id, value: 1, date:, type: "Appraisal" } }
+    assert_redirected_to account_path(@valuation.account)
+  end
+
+  test "should sync account after update" do
+    account = @valuation.account
+    date = @valuation.date
+    value = 2
+
+    account.sync
+
+    assert_changes("account.balance_on(date)", value) do
+      patch valuation_url(@valuation), params: { valuation: { account_id: @valuation.account_id, value:, date:, type: "Appraisal" } }
+      perform_enqueued_jobs
+    end
+  end
+
+  test "should do a partial account sync after update" do
+    account = @valuation.account
+    date = @valuation.date
+    value = 2
+
+    account.sync
+    account.balances.where(date: date - 2.day).update!(balance: 200)
+
+    assert_no_changes("account.balance_on(date - 2.day)") do
+      patch valuation_url(@valuation), params: { valuation: { account_id: @valuation.account_id, value:, date:, type: "Appraisal" } }
+      perform_enqueued_jobs
+    end
+  end
+
+  test "should destroy valuation" do
+    assert_difference("Valuation.count", -1) do
+      delete valuation_url(@valuation)
+    end
+
+    assert_redirected_to account_path(@account)
+  end
+
+  test "should sync account after destroy" do
+    account = @valuation.account
+    date = @valuation.date
+    value = @account.balance
+    account.sync
+
+    assert_changes("@account.balance_on(date)", to: 19700) do
+      delete valuation_url(@valuation)
+      perform_enqueued_jobs
+    end
+  end
+
+  test "should do a partial account sync after destroy" do
+    account = @valuation.account
+    date = @valuation.date
+
+    account.sync
+    account.balances.where(date: date - 10.day).update!(balance: 200)
+
+    assert_no_changes("account.balance_on(date - 10.day)") do
+      delete valuation_url(@valuation)
+      perform_enqueued_jobs
+    end
+  end
```

Comment:
So looking through the tests outlined in `valuations_controller_test` and `transactions_controller_test`, I think the test cases themselves are useful to have, but they probably don't belong in a controller test.  It looks like the questions we're asking here are:

1. Did the job happen?
2. Did the job happen with correct start date?
3. Did running the job result in the correct end state?

I think the first and the second assertions are correct to be here in the controllers, but the third question should probably be tested in `syncable_test` as it asserts against the logic that is handled by the `Account` model.

We could probably consolidate all the controller tests so that for each CRUD controller action, we assert that:

1. The resource was edited/created/deleted 
2. The sync job was enqueued with the correct params (we can use [assert_enqueued_with](https://guides.rubyonrails.org/testing.html#testing-jobs-in-context) for this).

That way, we're only testing the responsibility of the controller, which is to ensure the job is successfully enqueued with the correct args.

Reply:
Thanks for the great feedback, I agree!

Reply:
I've updated the tests